From 4a209977a61d156433dddbf21bddc0b1e89098e1 Mon Sep 17 00:00:00 2001
From: Gwenole Beauchesne <gwenole.beauchesne@intel.com>
Date: Mon, 28 Apr 2014 17:17:04 +0200
Subject: [PATCH 2/3] h264parse: fix build with GStreamer 1.2.

---
 gst/vaapi/gsth264parse.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/gst/vaapi/gsth264parse.c b/gst/vaapi/gsth264parse.c
index 4246b6e..7c970ee 100644
--- a/gst/vaapi/gsth264parse.c
+++ b/gst/vaapi/gsth264parse.c
@@ -148,7 +148,9 @@ gst_h264_parse_init (GstH264Parse * h264parse)
 {
   h264parse->frame_out = gst_adapter_new ();
   gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (h264parse), FALSE);
+#if GST_CHECK_VERSION(1,3,0)
   GST_PAD_SET_ACCEPT_INTERSECT (GST_BASE_PARSE_SINK_PAD (h264parse));
+#endif
 }
 
 
-- 
1.7.9.5

