From 06ab8f433c02590b63e8afdf394a2a1caf570fad Mon Sep 17 00:00:00 2001
From: Sreerenj Balachandran <sreerenj.balachandran@intel.com>
Date: Thu, 16 Apr 2015 18:04:53 +0300
Subject: [PATCH 5/6] h265parse: include gstvaapiparse.h

---
 gst/vaapi/gsth265parse.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gst/vaapi/gsth265parse.c b/gst/vaapi/gsth265parse.c
index d649681..5f65ab6 100644
--- a/gst/vaapi/gsth265parse.c
+++ b/gst/vaapi/gsth265parse.c
@@ -22,6 +22,7 @@
 #  include "config.h"
 #endif

+#include "gstvaapiparse.h"
 #include <gst/base/base.h>
 #include <gst/pbutils/pbutils.h>
 #include <gst/video/video.h>
@@ -100,7 +101,7 @@ gst_h265_parse_class_init (GstH265ParseClass * klass)
   GstBaseParseClass *parse_class = GST_BASE_PARSE_CLASS (klass);
   GstElementClass *gstelement_class = GST_ELEMENT_CLASS (klass);

-  GST_DEBUG_CATEGORY_INIT (h265_parse_debug, "h265parse", 0, "h265 parser");
+  GST_DEBUG_CATEGORY_INIT (h265_parse_debug, "vaapiparse_h265", 0, "h265 parser");

   gobject_class->finalize = gst_h265_parse_finalize;
   gobject_class->set_property = gst_h265_parse_set_property;
--
2.5.1
