--- po/Makefile.in.in.orig	2006-01-07 12:03:45.000000000 +0100
+++ po/Makefile.in.in	2006-01-07 12:04:23.000000000 +0100
@@ -11,6 +11,9 @@
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
 
+# thomas: add GETTEXT_PACKAGE substitution as used in Makevars
+GETTEXT_PACKAGE = @GETTEXT_PACKAGE@
+
 SHELL = /bin/sh
 @SET_MAKE@
 
@@ -305,7 +308,9 @@
 update-gmo: Makefile $(GMOFILES)
 	@:
 
-Makefile: Makefile.in.in $(top_builddir)/config.status POTFILES.in
+# thomas: add LINGUAS as a dependency so that the Makefile gets rebuilt
+# properly when we add languages
+Makefile: Makefile.in.in $(top_builddir)/config.status POTFILES.in LINGUAS
 	cd $(top_builddir) \
 	  && CONFIG_FILES=$(subdir)/$@.in CONFIG_HEADERS= \
 	       $(SHELL) ./config.status
