diff -ur ffmpeg-vanilla/libavformat/allformats.c ffmpeg/libavformat/allformats.c
--- ffmpeg-vanilla/libavformat/allformats.c	Thu May 29 18:34:46 2003
+++ ffmpeg/libavformat/allformats.c	Thu May 29 19:02:01 2003
@@ -40,13 +40,13 @@
 #endif
     avienc_init();
     avidec_init();
-    wav_init();
+    avf_wav_init();
     swf_init();
     au_init();
     gif_init();
     mov_init();
     jpeg_init();
-    dv_init();
+    avf_dv_init();
     fourxm_init();
 
     av_register_output_format(&yuv4mpegpipe_oformat);
diff -ur ffmpeg-vanilla/libavformat/avformat.h ffmpeg/libavformat/avformat.h
--- ffmpeg-vanilla/libavformat/avformat.h	Thu May 29 18:34:46 2003
+++ ffmpeg/libavformat/avformat.h	Thu May 29 19:02:09 2003
@@ -311,7 +311,7 @@
 int au_init(void);
 
 /* wav.c */
-int wav_init(void);
+int avf_wav_init(void);
 
 /* raw.c */
 int raw_init(void);
@@ -320,7 +320,7 @@
 int ogg_init(void);
 
 /* dv.c */
-int dv_init(void);
+int avf_dv_init(void);
 
 /* ffm.c */
 int ffm_init(void);
diff -ur ffmpeg-vanilla/libavformat/dv.c ffmpeg/libavformat/dv.c
--- ffmpeg-vanilla/libavformat/dv.c	Thu May 29 18:34:46 2003
+++ ffmpeg/libavformat/dv.c	Thu May 29 19:01:16 2003
@@ -138,7 +138,7 @@
     dv_write_trailer,
 };
 
-int dv_init(void)
+int avf_dv_init(void)
 {
     av_register_input_format(&dv_iformat);
     av_register_output_format(&dv_oformat);
diff -ur ffmpeg-vanilla/libavformat/wav.c ffmpeg/libavformat/wav.c
--- ffmpeg-vanilla/libavformat/wav.c	Thu May 29 18:34:46 2003
+++ ffmpeg/libavformat/wav.c	Thu May 29 19:02:32 2003
@@ -334,7 +334,7 @@
     wav_write_trailer,
 };
 
-int wav_init(void)
+int avf_wav_init(void)
 {
     av_register_input_format(&wav_iformat);
     av_register_output_format(&wav_oformat);
