diff -aur gstreamer-0.10.3-orig/gst/gstbin.h gstreamer-0.10.3/gst/gstbin.h
--- gstreamer-0.10.3-orig/gst/gstbin.h	2006-01-26 04:38:23.000000000 -0800
+++ gstreamer-0.10.3/gst/gstbin.h	2006-02-18 22:11:00.000000000 -0800
@@ -49,7 +49,7 @@
  */
 typedef enum {
   /* padding */
-  GST_BIN_FLAG_LAST		= (GST_ELEMENT_FLAG_LAST << 5)
+  GST_BIN_FLAG_LAST		= GST_ELEMENT_FLAG_LAST << 5
 } GstBinFlags;
 
 typedef struct _GstBin GstBin;
diff -aur gstreamer-0.10.3-orig/gst/gstbus.h gstreamer-0.10.3/gst/gstbus.h
--- gstreamer-0.10.3-orig/gst/gstbus.h	2006-01-21 06:07:10.000000000 -0800
+++ gstreamer-0.10.3/gst/gstbus.h	2006-02-18 22:11:27.000000000 -0800
@@ -47,9 +47,9 @@
  * The standard flags that a bus may have.
  */
 typedef enum {
-  GST_BUS_FLUSHING      = (GST_OBJECT_FLAG_LAST << 0),
+  GST_BUS_FLUSHING      = GST_OBJECT_FLAG_LAST << 0,
   /* padding */
-  GST_BUS_FLAG_LAST     = (GST_OBJECT_FLAG_LAST << 1)
+  GST_BUS_FLAG_LAST     = GST_OBJECT_FLAG_LAST << 1
 } GstBusFlags;
 
 /**
diff -aur gstreamer-0.10.3-orig/gst/gstelement.h gstreamer-0.10.3/gst/gstelement.h
--- gstreamer-0.10.3-orig/gst/gstelement.h	2005-11-29 10:57:59.000000000 -0800
+++ gstreamer-0.10.3/gst/gstelement.h	2006-02-18 22:25:06.000000000 -0800
@@ -425,7 +425,6 @@
 {
   GstObjectClass         parent_class;
 
-  /*< public >*/
   /* the element details */
   GstElementDetails	 details;
 
@@ -437,13 +436,11 @@
   gint                   numpadtemplates;
   guint32                pad_templ_cookie;
 
-  /*< private >*/
   /* signal callbacks */
   void (*pad_added)	(GstElement *element, GstPad *pad);
   void (*pad_removed)	(GstElement *element, GstPad *pad);
   void (*no_more_pads)	(GstElement *element);
 
-  /*< public >*/
   /* virtual methods for subclasses */
 
   /* request/release pads */
@@ -452,7 +449,7 @@
 
   /* state changes */
   GstStateChangeReturn (*get_state)		(GstElement * element, GstState * state,
-						 GstState * pending, GstClockTime timeout);
+			       		 GstState * pending, GstClockTime timeout);
   GstStateChangeReturn (*set_state)		(GstElement *element, GstState state);
   GstStateChangeReturn (*change_state)		(GstElement *element, GstStateChange transition);
 
@@ -473,7 +470,6 @@
   const GstQueryType*	(*get_query_types)	(GstElement *element);
   gboolean		(*query)		(GstElement *element, GstQuery *query);
 
-  /*< private >*/
   gpointer _gst_reserved[GST_PADDING];
 };
 
diff -aur gstreamer-0.10.3-orig/gst/gstindex.h gstreamer-0.10.3/gst/gstindex.h
--- gstreamer-0.10.3-orig/gst/gstindex.h	2006-01-19 00:38:56.000000000 -0800
+++ gstreamer-0.10.3/gst/gstindex.h	2006-02-18 22:11:57.000000000 -0800
@@ -149,11 +149,11 @@
  */
 typedef enum {
   GST_ASSOCIATION_FLAG_NONE 	  = 0,
-  GST_ASSOCIATION_FLAG_KEY_UNIT   = (1 << 0),
-  GST_ASSOCIATION_FLAG_DELTA_UNIT = (1 << 1),
+  GST_ASSOCIATION_FLAG_KEY_UNIT   = 1 << 0,
+  GST_ASSOCIATION_FLAG_DELTA_UNIT = 1 << 1,
 
   /* new flags should start here */
-  GST_ASSOCIATION_FLAG_LAST	= (1 << 8)
+  GST_ASSOCIATION_FLAG_LAST	= 1 << 8
 } GstAssocFlags;
 
 /**
@@ -294,10 +294,10 @@
  * Flags for this index
  */
 typedef enum {
-  GST_INDEX_WRITABLE    = (GST_OBJECT_FLAG_LAST << 0),
-  GST_INDEX_READABLE    = (GST_OBJECT_FLAG_LAST << 1),
+  GST_INDEX_WRITABLE    = GST_OBJECT_FLAG_LAST << 0,
+  GST_INDEX_READABLE    = GST_OBJECT_FLAG_LAST << 1,
 
-  GST_INDEX_FLAG_LAST   = (GST_OBJECT_FLAG_LAST << 8)
+  GST_INDEX_FLAG_LAST   = GST_OBJECT_FLAG_LAST << 8
 } GstIndexFlags;
 
 /**
diff -aur gstreamer-0.10.3-orig/gst/gstpad.h gstreamer-0.10.3/gst/gstpad.h
--- gstreamer-0.10.3-orig/gst/gstpad.h	2005-12-07 05:08:28.000000000 -0800
+++ gstreamer-0.10.3/gst/gstpad.h	2006-02-18 22:12:29.000000000 -0800
@@ -416,12 +416,12 @@
  * Pad state flags
  */
 typedef enum {
-  GST_PAD_BLOCKED       = (GST_OBJECT_FLAG_LAST << 0),
-  GST_PAD_FLUSHING      = (GST_OBJECT_FLAG_LAST << 1),
-  GST_PAD_IN_GETCAPS    = (GST_OBJECT_FLAG_LAST << 2),
-  GST_PAD_IN_SETCAPS    = (GST_OBJECT_FLAG_LAST << 3),
+  GST_PAD_BLOCKED       = GST_OBJECT_FLAG_LAST << 0,
+  GST_PAD_FLUSHING      = GST_OBJECT_FLAG_LAST << 1,
+  GST_PAD_IN_GETCAPS    = GST_OBJECT_FLAG_LAST << 2,
+  GST_PAD_IN_SETCAPS    = GST_OBJECT_FLAG_LAST << 3,
   /* padding */
-  GST_PAD_FLAG_LAST     = (GST_OBJECT_FLAG_LAST << 8)
+  GST_PAD_FLAG_LAST     = GST_OBJECT_FLAG_LAST << 8
 } GstPadFlags;
 
 /* FIXME: this awful circular dependency need to be resolved properly (see padtemplate.h) */
diff -aur gstreamer-0.10.3-orig/gst/gstpadtemplate.h gstreamer-0.10.3/gst/gstpadtemplate.h
--- gstreamer-0.10.3-orig/gst/gstpadtemplate.h	2005-12-21 08:13:56.000000000 -0800
+++ gstreamer-0.10.3/gst/gstpadtemplate.h	2006-02-18 22:12:50.000000000 -0800
@@ -103,9 +103,9 @@
  * Flags for the padtemplate
  */
 typedef enum {
-  GST_PAD_TEMPLATE_FIXED        = (GST_OBJECT_FLAG_LAST << 0),
+  GST_PAD_TEMPLATE_FIXED        = GST_OBJECT_FLAG_LAST << 0,
   /* padding */
-  GST_PAD_TEMPLATE_FLAG_LAST    = (GST_OBJECT_FLAG_LAST << 4)
+  GST_PAD_TEMPLATE_FLAG_LAST    = GST_OBJECT_FLAG_LAST << 4
 } GstPadTemplateFlags;
 
 /**
